std/algorithm/comparison.d

```diff
@@ -1241,7 +1248,14 @@ MaxType!T max(T...)(T args)
 
     //Do the "max" proper with a and b
     import std.functional : lessThan;
-    immutable chooseB = lessThan!(T0, T1)(a, b);
+    import std.math : isNaN;
+    static if (propagateNaN && isFloatingPoint!T1)
+        immutable chooseB = lessThan(a, b) || isNaN(b);
+    else static if (!propagateNaN && isFloatingPoint!T0)
```

Comment:
!propagateNaN is means “choose a value that is not NaN if one of them
is”, so there's a special case to handle that.  Or do you suggest
testing against PropagateNaN.no instead?